src/main/groovy/pl/allegro/tech/build/axion/release/domain/SnapshotDependenciesChecker.groovy

```diff
@@ -0,0 +1,27 @@
+package pl.allegro.tech.build.axion.release.domain
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Dependency
+
+class SnapshotDependenciesChecker {
+
+    Collection<String> snapshotVersions(Project project) {
+        def projectVersions = project.allprojects.collect {toFullVersion(it)}
+        def allDependenciesVersions = project.allprojects.collect {
+            it.configurations.collect { config ->
+                config.allDependencies.findAll {isSnapshot(it)}.collect {toFullVersion(it)}
+
+            }
+        }.flatten().unique()
+        allDependenciesVersions.removeAll(projectVersions)
+        return allDependenciesVersions
+    }
+
+    boolean isSnapshot(Dependency dependency) {
+        dependency.version?.endsWith("-SNAPSHOT")
+    }
+
+    String toFullVersion(it) {
+        "${it.group}:${it.name}:${it.version}"
```

Comment:
I know that in case of Gradle in many places you cannot be sure what object is inside, but here `Project` should be quite safe. IMHO this code would be more readable with less `def`s.